src/OrchardCore.Modules/OrchardCore.Users/Workflows/Drivers/SelectUsersInRoleTaskDisplayDriver.cs

```diff
@@ -0,0 +1,22 @@
+using OrchardCore.Users.Workflows.Activities;
+using OrchardCore.Users.Workflows.ViewModels;
+using OrchardCore.Workflows.Display;
+using OrchardCore.Workflows.Models;
+
+namespace OrchardCore.Users.Workflows.Drivers
+{
+    public class SelectUsersInRoleTaskDisplayDriver : ActivityDisplayDriver<SelectUsersInRoleTask, SelectUsersInRoleTaskViewModel>
+    {
+        protected override void EditActivity(SelectUsersInRoleTask activity, SelectUsersInRoleTaskViewModel model)
+        {
+            model.PropertyName = activity.PropertyName.Expression;
```

Comment:
I don't think you need PropertyName. All you should need is a list of roles

```
public RoleEntry[] Roles { get; set; } = Array.Empty<RoleEntry>();
```

Reply:
@elaurentin yes I am aware that you are creating two different activities. The `SelectUsersInRoleTask` should return a collection of User object instead of `userNames` or `emails` or,..... This would be much more useful.

To do this, you'll need a property key name (`OutputKeyName`) to have the users collections into so that the user can then use the collection to do anything they want (like get a semi-colon separated list). 

The idea of having different `OutputKeyName` values is that you can store different collections of users in the output. For example, `Workflow.Output("AdminUsers")` or `Workflow.Output("Users")`....

Now, on the next task like `SendEmailTask` you can loop over `Workflow.Output("Users")` using Liquid and create a semi-colon list of email as recipient. 

As for the checkbox list, on the UI, you would render the checkbox list as I showed above. Then you can convert the user selection to a string array of roles as json and save the json into the string.